app/models/provided/exchange_rate.rb

```diff
@@ -0,0 +1,22 @@
+class Provided::ExchangeRate
+  include Providable, Retryable
+
+  def initialize
+    @provider = exchange_rates_provider
```

Comment:
Hmm. If we unroll this - 

```
  def exchange_rates_provider
    self.class.exchange_rates_provider
  end
```

```
    def exchange_rates_provider
      provider_for :exchange_rates
    end
```

```
      def provider_for(concept)
        "Provider::#{provider_name(concept).camelize}"
          .constantize
          .new(provider_api_key(concept))
      end
```

```
      def provider_name(concept)
        providers_config[concept][:provider].presence_in(KNOWN_PROVIDERS) ||
          raise(ArgumentError, "Unknown provider for #{concept}")
      end
```

```
shared:
  exchange_rates:
    provider: <%= ENV.fetch "EXCHANGE_RATES_PROVIDER", "synth" %>
    key: <%= ENV.fetch "EXCHANGE_RATES_API_KEY", ENV["SYNTH_API_KEY"] %>
```

So all this... looks like it evaluates to `@provider = Providers::Synth.new`?

Reply:
That eliminates the two outermost layers of abstraction only, which are the thinnest wrappers.

Those were implemented on purpose. I think it's better for the caller to only know the name of a method rather than the name pf a module, a method, and a slightly magical symbol argument.

Personally not worried about those two thin wrappers. They're meant to provide a purposeful public API.